conda_env/cli/main.py

```diff
@@ -1,49 +1,28 @@
 # Copyright (C) 2012 Anaconda, Inc
 # SPDX-License-Identifier: BSD-3-Clause
-"""Entry point for all conda-env subcommands."""
+"""DEPRECATED: Use `conda.cli.main_env` instead.
+
+Entry point for all conda-env subcommands.
+"""
 import os
 import sys
 from importlib import import_module
 
-# pip_util.py import on_win from conda.exports
-# conda.exports resets the context
-# we need to import conda.exports here so that the context is not lost
-# when importing pip (and pip_util)
-import conda.exports  # noqa
 from conda.base.context import context
-from conda.cli.conda_argparse import ArgumentParser
 from conda.cli.main import init_loggers
+from conda.cli.main_env import configure_parser as create_parser  # noqa
+from conda.deprecations import deprecated
 from conda.exceptions import conda_exception_handler
 from conda.gateways.logging import initialize_logging
 
-from . import main_config, main_create, main_export, main_list, main_remove, main_update
+deprecated.module("23.9", "24.3", addendum="Use `conda.cli.main_env` instead.")
 
 
-# TODO: This belongs in a helper library somewhere
-# Note: This only works with `conda-env` as a sub-command.  If this gets
-# merged into conda-env, this needs to be adjusted.
 def show_help_on_empty_command():
     if len(sys.argv) == 1:  # sys.argv == ['/path/to/bin/conda-env']
         sys.argv.append("--help")
 
 
-def create_parser():
-    p = ArgumentParser()
-    sub_parsers = p.add_subparsers(
-        metavar="command",
-        dest="cmd",
-    )
-    main_create.configure_parser(sub_parsers)
-    main_export.configure_parser(sub_parsers)
-    main_list.configure_parser(sub_parsers)
-    main_remove.configure_parser(sub_parsers)
-    main_update.configure_parser(sub_parsers)
-    main_config.configure_parser(sub_parsers)
-
-    show_help_on_empty_command()
-    return p
-
-
```

Comment:
After some more thought I realize we haven't properly back ported here. In addition to the if-else clause added to `conda.cli.main_env.configure_parser` we also need to ensure that if a downstream project invokes `conda_env.cli.main.create_parser()` **with no arguments** that the function will succeed and return a valid `ArgumentParser`. Something like this:

```suggestion
def create_parser():
    return configure_parser(None)
```